src/vm/comsynchronizable.cpp

```diff
@@ -1537,15 +1537,56 @@ void QCALLTYPE ThreadNative::nativeInitCultureAccessors()
     END_QCALL;
 }
 
+typedef HRESULT(WINAPI *pfnSetThreadDescription)(HANDLE hThread, PCWSTR lpThreadDescription);
+extern pfnSetThreadDescription g_pfnSetThreadDescription;
+BOOL g_InitializeSetThreadDescriptionWasRun = FALSE;
+
+
+HRESULT WINAPI InitializeSetThreadDescription(HANDLE hThread, PCWSTR lpThreadDescription)
+{
+    pfnSetThreadDescription func = NULL;
+
+#ifndef FEATURE_PAL
+    if (!g_InitializeSetThreadDescriptionWasRun)
+    {
+        HMODULE hKernel32 = WszLoadLibrary(W("kernel32.dll"));
+        if (hKernel32 != NULL)
+        {
+            func = (pfnSetThreadDescription)GetProcAddress(hKernel32, "SetThreadDescription");
+        }
+        g_InitializeSetThreadDescriptionWasRun = TRUE;
```

Comment:
Yes that is the intention of sparing the GetProcAddress call. Apart from that it would be no different than an empty method. If func is null it will simply return S_OK which is fine for the caller if the feature is not present.